internal/api/handler.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+import (
+	"net/http"
+)
+
+type Handler struct {
+	store ReaderWriter
+}
+
+type Reader interface {
+	Get(metricType string, metricName string) (string, bool)
+}
+
+type Writer interface {
+	Add(metricType string, metricName string, metricValue string) error
+}
+
+type ReaderWriter interface {
+	Reader
+	Writer
+}
+
+func NewHandler(store ReaderWriter) *http.ServeMux {

```

Comment:
Мне кажется что фабричный метод `NewHandler` должен возвращать инстанс хэндлера.
```suggestion
func NewHandler(store ReaderWriter) *Handler {
```